src/pytorch_lightning/plugins/environments/azure_openmpi_environment.py

```diff
@@ -0,0 +1,77 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+
+
+class AzureOpenMPIEnvironment(ClusterEnvironment):
+    """Environment for an OpenMPI environment on Azure.
+
+    See Azure documentation here: https://docs.microsoft.com/en-us/azure/machine-learning/how-to-train-distributed-gpu#mpi
+    """
+
+    def __init__(self, devices: int = 1) -> None:
+        """devices : devices per node (same as trainer parameter)"""
+        super().__init__()
+        self.devices = devices
+
+    @property
+    def creates_processes_externally(self) -> bool:
+        """Return True if the cluster is managed (you don't launch processes yourself)"""
+        return True
+
+    def world_size(self) -> int:
+        return int(os.environ.get("OMPI_COMM_WORLD_SIZE"))
+
+    def set_world_size(self, size: int) -> None:
+        pass
+
+    def global_rank(self) -> int:
+        return int(os.environ.get("OMPI_COMM_WORLD_RANK"))
+
+    def set_global_rank(self, rank: int) -> None:
+        pass
+
+    def local_rank(self) -> int:
+        return int(os.environ.get("OMPI_COMM_WORLD_LOCAL_RANK"))
+
+    def node_rank(self) -> int:
+        # this may not exist, defaulting to 0
+        return int(os.environ.get("OMPI_COMM_WORLD_RANK", 0)) // int(self.devices)
+
+    @property
+    def main_address(self) -> str:
```

Comment:
Great! If we could follow the style of the other classes and move the properties to the top, followed by all methods, that would be nice!

Reply:
This is fixed now